File: 2047-find-a-peak-element-ii/find-a-peak-element-ii.js
var findPeakGrid = function(mat) {
    let i = 0
    let rows = mat.length 
    let j = 0
    let columns = mat[0].length 

    while (j < columns && i < rows) {
        if (i > 0 && mat[i][j] < mat[i - 1][j]) i -= 1
        else if (j < columns - 1 && mat[i][j] < mat[i][j + 1]) j += 1
        else if (i < rows - 1 && mat[i][j] < mat[i + 1][j]) i += 1
        else if (j > 0 && mat[i][j] < mat[i][j - 1]) j -= 1
        else return [i, j]
    }

};

File: 2047-find-a-peak-element-ii/find-a-peak-element-ii.py
class Solution:
    def findPeakGrid(self, mat: List[List[int]]) -> List[int]:
        rows = len(mat)
        columns = len(mat[0])
        i = 0
        j = 0

        while (i < rows and j < columns):
            cell = mat[i][j]

            if (i > 0 and cell <= mat[i - 1][j]):
                i -= 1
            elif (j < columns - 1 and cell <= mat[i][j + 1]):
                j += 1
            elif (i < rows - 1 and cell <= mat[i + 1][j]):
                i += 1
            elif (j > 0 and cell <= mat[i][j - 1]):
                j -= 1
            else:
                return [i, j]

        return [-1, -1]
        